symengine/basic-methods.inc

```diff
@@ -1,4 +1,8 @@
 class EvalRealDoubleVisitorFinal;
-#define SYMENGINE_INCLUDE_METHODS(SUFFIX) \
-virtual void accept(Visitor &v) const SUFFIX \
-virtual void accept(EvalRealDoubleVisitorFinal &v) const SUFFIX
+#define _SYMENGINE_INCLUDE_METHODS(PREFIX, SUFFIX) \
+PREFIX void accept(Visitor &v) const SUFFIX \
+PREFIX void accept(EvalRealDoubleVisitorFinal &v) const SUFFIX
+
+#define SYMENGINE_INCLUDE_METHODS_BASE() _SYMENGINE_INCLUDE_METHODS(virtual, = 0;)
+
+#define SYMENGINE_INCLUDE_METHODS_DERIVED() _SYMENGINE_INCLUDE_METHODS(, override;)
```

Comment:
Let's get rid of `_SYMENGINE_INCLUDE_METHODS` and expand the two methods so that it's easier to read. (Previous way was hard to read too)